### PR TITLE
[PROGRAMMES-7002] Removed A-Z Search

### PR DESCRIPTION
**Pull request by benolot**
Ticket: https://jira.dev.bbc.co.uk/browse/PROGRAMMES-7002

I just flat out removed it, iirc there was no plans to redirect to main BBC search.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by Tasiobg** [comment]
test this please

File: config/routes.yaml [hunk 1]
```diff
@@ -216,30 +216,21 @@ atoz_index:
     path: /programmes/a-z
     defaults: { _controller: App\Controller\Atoz\IndexController }
 
-atoz_list:
-    path: /programmes/a-z/{slice}
-    defaults: { _controller: App\Controller\Atoz\ListController }
-    requirements: { slice: 'all|player' }
-
 atoz_redirect_current_list:
```
Comment (Bobkidbob) [inline config/routes.yaml]: This can also go, it is an old slice that would've been handled by `atoz_list`.

File: config/routes.yaml [hunk 2]
```diff
@@ -216,30 +216,21 @@ atoz_index:
...
     defaults: { route: atoz_show, slice: "player", permanent: true }
+    requirements: { search: '^[a-zA-Z@]$' }
 
 atoz_show:
     path: /programmes/a-z/by/{search}/{slice}
     defaults: { _controller: App\Controller\Atoz\ShowController }
-    requirements: { slice: 'all|player' }
+    requirements: { slice: 'all|player',search: '^[a-zA-Z@]$' }
 
 atoz_redirect_current_show:
```
Comment (Bobkidbob) [inline config/routes.yaml]: If we're getting rid of old redirects this can probably go as well.

File: src/Controller/Atoz/ShowController.php [hunk 3]
```diff
@@ -36,76 +37,47 @@ public function __invoke(
...
-                        );
-                    } else {
-                        $results = [];
-                    }
-                    break;
-                default:
-                    $count = 0;
+        if (!preg_match('/^[a-zA-Z@]$/', $search)) {
+            // Throw a 404 if not a single letter or @ as that's a search and that's invalid now
+            throw new NotFoundHttpException();
```
Comment (Bobkidbob) [inline src/Controller/Atoz/ShowController.php]: Is this needed now that the routes have it set as a requirement?

Comment (benolot) [reply src/Controller/Atoz/ShowController.php]: Yeah it throws a 500 as this code makes sure it's single char or nothing (Rest of code assumes single char)

**Pull request merged**